<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BGCMSMainWin</class>
 <widget class="QMainWindow" name="BGCMSMainWin">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BlueGhost CMS</string>
  </property>
  <property name="windowIcon">
   <iconset resource="bgcms.qrc">
    <normaloff>:/imgs/appicon.png</normaloff>:/imgs/appicon.png</iconset>
  </property>
  <property name="0" stdset="0">
   <number>0</number>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QMdiArea" name="mdiArea">
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="tabsMovable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>文档(&amp;F)</string>
    </property>
    <addaction name="actionConnect_to_CMS"/>
    <addaction name="actiondisconnect"/>
   </widget>
   <widget class="QMenu" name="menu_Window">
    <property name="title">
     <string>窗口(&amp;W)</string>
    </property>
    <addaction name="actionWindowView"/>
    <addaction name="actionTabedView"/>
   </widget>
   <widget class="QMenu" name="ui_menuPlugins">
    <property name="title">
     <string>插件(&amp;P)</string>
    </property>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="ui_menuPlugins"/>
   <addaction name="menu_Window"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="ui_dwSubWinList">
   <property name="features">
    <set>QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>打开窗口列表</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="SubWinList" name="ui_lvSubWinList">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="ui_dwPlugins">
   <property name="minimumSize">
    <size>
     <width>150</width>
     <height>145</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>插件</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SetMinAndMaxSize</enum>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QToolBox" name="ui_tbxPlugins">
       <property name="tabSpacing">
        <number>0</number>
       </property>
       <widget class="QWidget" name="widget">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>150</width>
          <height>508</height>
         </rect>
        </property>
        <attribute name="label">
         <string>工具箱</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QScrollArea" name="scrollArea">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>148</width>
              <height>506</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="leftMargin">
              <number>2</number>
             </property>
             <property name="topMargin">
              <number>2</number>
             </property>
             <property name="rightMargin">
              <number>2</number>
             </property>
             <property name="bottomMargin">
              <number>2</number>
             </property>
             <item alignment="Qt::AlignTop">
              <widget class="QWidget" name="ui_widPluginsToolbox" native="true">
               <layout class="QVBoxLayout" name="ui_vlPlugins">
                <property name="spacing">
                 <number>0</number>
                </property>
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionConnect_to_CMS">
   <property name="text">
    <string>连接到CMS</string>
   </property>
  </action>
  <action name="actiondisconnect">
   <property name="text">
    <string>断开连接</string>
   </property>
  </action>
  <action name="actionWindowView">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>WindowView</string>
   </property>
   <property name="value" stdset="0">
    <number>0</number>
   </property>
  </action>
  <action name="actionTabedView">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>TabedView</string>
   </property>
   <property name="value" stdset="0">
    <number>1</number>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SubWinList</class>
   <extends>QListView</extends>
   <header location="global">subwinlist.h</header>
   <slots>
    <signal>closeSubWin(int)</signal>
    <signal>activeSubWin(int)</signal>
    <slot>addEditorItem(QVariantMap)</slot>
    <slot>removeItem(int)</slot>
    <slot>addPluginItem(int,QString)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="bgcms.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionConnect_to_CMS</sender>
   <signal>triggered()</signal>
   <receiver>BGCMSMainWin</receiver>
   <slot>connectToCMS()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BGCMSMainWin</sender>
   <signal>editorOpened(QVariantMap)</signal>
   <receiver>ui_lvSubWinList</receiver>
   <slot>addEditorItem(QVariantMap)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BGCMSMainWin</sender>
   <signal>subWinClosed(int)</signal>
   <receiver>ui_lvSubWinList</receiver>
   <slot>removeItem(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mdiArea</sender>
   <signal>subWindowActivated(QMdiSubWindow*)</signal>
   <receiver>ui_lvSubWinList</receiver>
   <slot>activedSubWinChanged(QMdiSubWindow*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>377</x>
     <y>114</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ui_lvSubWinList</sender>
   <signal>activeSubWin(int)</signal>
   <receiver>BGCMSMainWin</receiver>
   <slot>activeSubWin(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>250</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actiondisconnect</sender>
   <signal>triggered()</signal>
   <receiver>BGCMSMainWin</receiver>
   <slot>disconnectFromCMS()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ui_lvSubWinList</sender>
   <signal>closeSubWin(int)</signal>
   <receiver>BGCMSMainWin</receiver>
   <slot>closeSubWin(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BGCMSMainWin</sender>
   <signal>pluginSubWinOpened(int,QString)</signal>
   <receiver>ui_lvSubWinList</receiver>
   <slot>addPluginItem(int,QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>editorOpened(QVariantMap)</signal>
  <signal>subWinClosed(int)</signal>
  <signal>pluginSubWinOpened(int,QString)</signal>
  <slot>connectToCMS()</slot>
  <slot>activeSubWin(int)</slot>
  <slot>disconnectFromCMS()</slot>
  <slot>closeSubWin(int)</slot>
 </slots>
</ui>
